File: app/src/main/kotlin/com/spacex/ui/rocketDetails/launches/RocketLaunchesHeaderVH.java
package com.spacex.ui.rocketDetails.launches;

import androidx.annotation.NonNull;
import com.spacex.ui.databinding.RocketLaunchesHeaderVhBinding;

class RocketLaunchesHeaderVH extends RocketLaunchesItemVH {

  private final RocketLaunchesHeaderVhBinding mViewBinding;

  RocketLaunchesHeaderVH(@NonNull final RocketLaunchesHeaderVhBinding binding) {
    super(binding.getRoot());
    mViewBinding = binding;
  }

  void onBind(@NonNull final RocketLaunchHeaderUIM rocketLaunchesHeaderUIM) {
    mViewBinding.setUIM(new RocketLaunchesHeaderUIMDecorator(rocketLaunchesHeaderUIM));
    mViewBinding.executePendingBindings();
  }
}

File: app/src/main/kotlin/com/spacex/ui/rocketDetails/launches/RocketLaunchesHeaderVH.kt
package com.spacex.ui.rocketDetails.launches

import com.spacex.ui.databinding.RocketLaunchesHeaderVhBinding

internal class RocketLaunchesHeaderVH(
        private val mViewBinding: RocketLaunchesHeaderVhBinding
) : RocketLaunchesItemVH(mViewBinding.root) {

    fun onBind(rocketLaunchesHeaderUIM: RocketLaunchHeaderUIM) {
        mViewBinding.uim = RocketLaunchesHeaderUIMDecorator(rocketLaunchesHeaderUIM)
        mViewBinding.executePendingBindings()
    }
}
